src/encoder.rs

```diff
@@ -2075,6 +2084,9 @@ pub fn encode_block_with_modes<T: Pixel>(
   let mode_context =
     cw.find_mvrefs(tile_bo, ref_frames, &mut mv_stack, bsize, fi, is_compound);
 
+  if !mode_decision.skip && !mode_decision.has_coeff {
```

Comment:
Does the !mode_decision.skip condition do anything?

Reply:
Yes, in mode_decision_rdo() we test two skip modes, true and false, and that is kept by  mode_decision.skip. After mode_decision is done, and if mode_decision.skip is false but mode_decision.has_coeff is false, this means the case that there is no non-zero coeff while skip is set false. This condition can be checked possibly at earlier stage, but I ended yup at this place.

Reply:
And I guess you probably thought that has_coeff is always false if mode_decision.skip is true. Yes, that is right.